internal/runtime/client/client_test.go

```diff
@@ -656,8 +726,144 @@ func TestClient_CallExtension(t *testing.T) {
+		{
+			name:              "Aggregate response if there is only one response",
+			aggregateResponse: fakeResponse(),
+			responses: []runtimehooksv1.ResponseObject{
+				fakeResponse(),
+			},
+			want: fakeResponse(),
+		},
+		{
+			name:              "Aggregate retry response if there is only one response",
+			aggregateResponse: fakeRetryableResponse(0),
+			responses: []runtimehooksv1.ResponseObject{
+				fakeRetryableResponse(5),
+			},
+			want: fakeRetryableResponse(5),
+		},
+		{
+			name:              "Aggregate retry responses to lowest non-zero retryAfterSeconds value",
+			aggregateResponse: fakeRetryableResponse(0),
+			responses: []runtimehooksv1.ResponseObject{
+				fakeRetryableResponse(0),
+				fakeRetryableResponse(1),
+				fakeRetryableResponse(5),
+				fakeRetryableResponse(4),
+				fakeRetryableResponse(3),
+			},
+			want: fakeRetryableResponse(1),
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			aggregateSuccessfulResponses(tt.aggregateResponse, tt.responses)
+
+			if !reflect.DeepEqual(tt.aggregateResponse, tt.want) {
+				t.Errorf("aggregateSuccessfulResponses() got = %v, want %v", tt.aggregateResponse, tt.want)
+			}
+		})
+	}
+}
+
 type testServerConfig struct {
-	start              bool
+	start     bool
+	responses map[string]testServerResponse
+	hostPort  string
+}
+
+type testServerResponse struct {
 	response           runtime.Object
 	responseStatusCode int
 }
+
+func response(status runtimehooksv1.ResponseStatus) testServerResponse {
+	return testServerResponse{
+		response: &fakev1alpha1.FakeResponse{
+			CommonResponse: runtimehooksv1.CommonResponse{
+				Status: status,
+			},
+		},
+		responseStatusCode: http.StatusOK,
+	}
+}
+
+func createTestServer(g *WithT, server testServerConfig) *httptest.Server {
+	l, err := net.Listen("tcp", server.hostPort)
+	g.Expect(err).NotTo(HaveOccurred())
+
+	mux := http.NewServeMux()
+	mux.HandleFunc("/", func(w http.ResponseWriter, r *http.Request) {
+		// Write the response for the first match in tt.testServer.responses.
+		for pathRegex, resp := range server.responses {
+			if !regexp.MustCompile(pathRegex).MatchString(r.URL.Path) {
+				continue
+			}
+
+			respBody, err := json.Marshal(resp.response)
+			if err != nil {
+				panic(err)
+			}
+			w.WriteHeader(resp.responseStatusCode)
+			_, _ = w.Write(respBody)
+			return
+		}
+
+		// Otherwise write a 404.
+		w.WriteHeader(http.StatusNotFound)
+	})
+	srv := httptest.NewUnstartedServer(mux)
+	// NewUnstartedServer creates a listener. Close that listener and replace
+	// with the one we created.
+	g.Expect(srv.Listener.Close()).To(Succeed())
+	srv.Listener = l
+
+	return srv
+}
+
+func registry(configs []runtimev1.ExtensionConfig) runtimeregistry.ExtensionRegistry {
+	registry := runtimeregistry.New()
+	err := registry.WarmUp(&runtimev1.ExtensionConfigList{
+		Items: configs,
+	})
+	if err != nil {
+		panic(err)
+	}
+	return registry
+}
+
+func fakeResponse() *fakev1alpha1.FakeResponse {
```

Comment:
```suggestion
func fakeSuccessResponse() *fakev1alpha1.FakeResponse {
```

nit. Feel free to resolve if you don't want to rename it :)

Reply:
Good idea!